contracts/traderjoe/FarmLens.sol

```diff
@@ -0,0 +1,217 @@
+    }
+
+    // Need to be aware of decimals here, not always 18, it depends on the token
+    function getPriceInAvax(address tokenAddress) public view returns (uint256) {
+        if (tokenAddress == wavax) {
+            return 1e18;
+        }
+
+        IJoePair pair = IJoePair(joeFactory.getPair(tokenAddress, wavax));
+
+        (uint256 reserve0, uint256 reserve1, ) = pair.getReserves();
+        address token0Address = pair.token0();
+        address token1Address = pair.token1();
+
+        if (token0Address == wavax) {
+            reserve1 = reserve1.mul(_tokenDecimalsMultiplier(token1Address)); // 18
+            return (reserve0.mul(1e18)) / reserve1; // 18
+        } else {
+            reserve0 = reserve0.mul(_tokenDecimalsMultiplier(token0Address)); // 18
+            return (reserve1.mul(1e18)) / reserve0; // 18
+        }
+    }
+
+    function _tokenDecimalsMultiplier(address tokenAddress) private pure returns (uint256) {
+        uint256 decimalsNeeded = 18 - IJoeERC20(tokenAddress).decimals();
+        return 1 * (10**decimalsNeeded);
+    }
+
+    function getReserveUSD(IJoePair pair) public view returns (uint256) {
+        address token0Address = pair.token0();
+        address token1Address = pair.token1();
+
+        (uint256 reserve0, uint256 reserve1, ) = pair.getReserves();
+
+        reserve0 = reserve0.mul(_tokenDecimalsMultiplier(token0Address)); // 18
+        reserve1 = reserve1.mul(_tokenDecimalsMultiplier(token1Address)); // 18
+
+        uint256 token0PriceInAvax = getPriceInAvax(token0Address); // 18
+        uint256 token1PriceInAvax = getPriceInAvax(token1Address); // 18
+        uint256 reserve0Avax = reserve0.mul(token0PriceInAvax); // 36;
+        uint256 reserve1Avax = reserve1.mul(token1PriceInAvax); // 36;
+        uint256 reserveAVAX = (reserve0Avax.add(reserve1Avax)) / 1e18; // 18
+        uint256 reserveUSD = (reserveAVAX.mul(getAvaxPrice())) / 1e18; // 18
+
+        return reserveUSD; // 18
+    }
+
+    struct FarmPair {
+        address lpAddress;
+        address token0Address;
+        address token1Address;
+        string token0Symbol;
+        string token1Symbol;
+        uint256 reserveUSD;
+        uint256 totalSupply;
+        address chefAddress;
+        uint256 chefBalance;
+        uint256 chefTotalAlloc;
+        uint256 chefJoePerSec;
+    }
+
+    function getFarmPairs(address[] calldata pairAddresses, address chefAddress)
+        public
+        view
+        returns (FarmPair[] memory)
+    {
+        uint256 farmCount;
+        uint256 farmPairIndex = 0;
+        // get count of farm pairs that this masterChef owns, needed due to solidity lacking dynamic memory array support
+        for (uint256 i = 0; i < pairAddresses.length; i++) {
+            IJoePair lpToken = IJoePair(pairAddresses[i]);
+            uint256 balance = lpToken.balanceOf(chefAddress);
+            if (balance > 0) {
+                farmCount++;
+            }
+        }
```

Comment:
Not very accurate to do this. Why? Because naiive users like to send tokens to contract by accident. 

To get number of pools on MC, simply call `chef.poolLength()` - you will need to update `IMasterChef` too

Reply:
ahh ok makes sense, good call

Reply:
ahh ok makes sense, good call